src/zarr/core/common.py

```diff
@@ -150,9 +150,11 @@ def parse_shapelike(data: int | Iterable[int]) -> tuple[int, ...]:
     return data_tuple
 
 
-def parse_fill_value(data: Any) -> Any:
+def parse_fill_value(fill_value: Any, dtype: Any, zarr_format: ZarrFormat) -> Any:
+    if zarr_format == 2 and (dtype is str or dtype == "str") and fill_value == 0:
```

Comment:
I think this needs at least a comment explaining why we are doing this. what was zarr-python 2's behavior in this situation?